### PR TITLE
Tag 28

### PR DESCRIPTION
**Pull request by kacperantoniak**
Implementation of "random" data injection to database

### CODE REVIEW

File: TagsterWebAPI/src/Tagster.Application/Services/IAdminService.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Tagster.Application.Services
+{
+    public interface IAdminService
+    {
+        public Task CreateFakeData(int profilesCount, int maxTagsPerProfile);
```
Comment (flapek) [inline TagsterWebAPI/src/Tagster.Application/Services/IAdminService.cs:11]: Rename `CreateFakeData` => `CreateFakeDataAsync`

File: TagsterWebAPI/src/Tagster.Application/Utils/FakeDataUtils.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Tagster.Application.Utils
+{
+    public class FakeDataUtils
```
Comment (flapek) [inline TagsterWebAPI/src/Tagster.Application/Utils/FakeDataUtils.cs:9]: Better way is to create JSON file which store data this class

File: TagsterWebAPI/src/TagsterWebAPI/Controllers/AdminController.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    [Route("admin")]
+    public class AdminController : ControllerBase
+    {
+        private readonly IAdminService _adminService;
+
+        public AdminController(IAdminService adminService)
+            => _adminService = adminService;
+
+        [HttpGet]
+        [Route("generate-fake-data/p{profilesCount}/max{maxTagsPerProfile}")]//add variables to route! (profilesCount, maxTagsPerProfile)
```
Comment (flapek) [inline TagsterWebAPI/src/TagsterWebAPI/Controllers/AdminController.cs:21]: Change code comments to documentations comment and add more details to it

File: TagsterWebAPI/src/Tagster.Infrastructure/Services/AdminService.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+                for (int j = 0; j<rand.Next(maxTagsPerProfile); j++)
+                {
+                    string tagName = FakeDataUtils.Tags[rand.Next(FakeDataUtils.Tags.Length)];
+                    Tag tag = new Tag();
+                    tag.TagName = tagName;
+                    tags.Add(tag);
+                }
+
+                var profile = new Profile
```
Comment (flapek) [inline TagsterWebAPI/src/Tagster.Infrastructure/Services/AdminService.cs:41]: Use Factory pattern to create profile

File: TagsterWebAPI/src/Tagster.Infrastructure/Services/AdminService.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        public async Task CreateFakeData(int profilesCount, int maxTagsPerProfile)
+        {
+            Random rand = new();
+            for (int i=0; i<profilesCount; i++) 
+            {
+                var name = FakeDataUtils.Names[rand.Next(FakeDataUtils.Names.Length)];
+
+                var surname = FakeDataUtils.Surnames[rand.Next(FakeDataUtils.Surnames.Length)];
+
+                var tags = new List<Tag>();
```
Comment (flapek) [inline TagsterWebAPI/src/Tagster.Infrastructure/Services/AdminService.cs:31]: Use Factory pattern to create list of tags

File: TagsterWebAPI/src/Tagster.Infrastructure/Services/AdminService.cs [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+                var profile = new Profile
+                {
+                    Href = Guid.NewGuid().ToString(),
+                    LastName = surname,
+                    Name = name,
+                    ProfileTags = tags
+                };
+
+                await _context.Profiles.AddAsync(profile);
```
Comment (flapek) [inline TagsterWebAPI/src/Tagster.Infrastructure/Services/AdminService.cs:49]: You should add list of profiles to context outside loop

**Review by flapek — CHANGES_REQUESTED** [review]

File: TagsterWebAPI/src/TagsterWebAPI/Controllers/AdminController.cs (hunk 3 above)
Comment (kacperantoniak) [reply TagsterWebAPI/src/TagsterWebAPI/Controllers/AdminController.cs:21]: forgot to delete it, it was my "to do" kind of thing

**Review by flapek — APPROVED** [review]

**Pull request merged**